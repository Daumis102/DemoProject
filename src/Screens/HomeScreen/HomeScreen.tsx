import React from 'react';
import {Text, View} from 'react-native';
import useOrientationStyle from '../../hooks/useOrientationStyle';
import {FormatedScrollView, FormatedView} from '../../SharedComponents';
import styles from './style';

export const HomeScreen = () => {
  const {style} = useOrientationStyle(styles.portrait, styles.landscape);
  return (
    <View style={style.container}>
      <FormatedView style={style.headerView}>
        <Text style={style.headerText}>Hello</Text>
      </FormatedView>
      <FormatedScrollView style={style.scrollView}>
        <View style={style.scrollViewItem}>
          <Text>Scrollable</Text>
        </View>
        <View style={style.scrollViewItem}>
          <Text>Scrollable</Text>
        </View>
        <View style={style.scrollViewItem}>
          <Text>Scrollable</Text>
        </View>
        <View style={style.scrollViewItem}>
          <Text>Scrollable</Text>
        </View>
        <View style={style.scrollViewItem}>
          <Text>Scrollable</Text>
        </View>
      </FormatedScrollView>
    </View>
  );
};
