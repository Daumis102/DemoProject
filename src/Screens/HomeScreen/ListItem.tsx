import React from 'react';
import {Text, View} from 'react-native';

type Props = {
  text: string;
};

export const ListItem = ({text}: Props) => {
  return (
    <View>
      <Text>{text}</Text>
    </View>
  );
};
