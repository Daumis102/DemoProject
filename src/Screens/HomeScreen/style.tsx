import {StyleSheet} from 'react-native';

const baseStyle = StyleSheet.create({});

const portrait = StyleSheet.create({
  ...baseStyle,
  headerView: {flexBasis: 100, flexGrow: 1},
  headerText: {
    backgroundColor: 'blue',
    textAlign: 'center',
    textAlignVertical: 'center',
    flex: 1,
  },
  scrollView: {
    flexBasis: 50,
    backgroundColor: 'blue',
    flexGrow: 2,
    borderWidth: 1,
  },
  scrollViewItem: {
    flexBasis: 50,
    flexGrow: 1.3,
    backgroundColor: 'red',
  },
  container: {
    height: 300,
    borderWidth: 1,
    flexDirection: 'column',
  },
});

const landscape = StyleSheet.create({
  ...baseStyle,
  headerView: {flexBasis: 100, flexGrow: 1},
  headerText: {
    textAlign: 'center',
    textAlignVertical: 'center',
    flex: 1,
  },
  scrollView: {
    flexBasis: 50,
    flexGrow: 2,
    borderWidth: 1,
  },
  scrollViewItem: {
    flexBasis: 50,
    flexGrow: 1.3,
    backgroundColor: 'red',
  },
  container: {
    height: 300,
    borderWidth: 1,
    flexDirection: 'row',
  },
});

const styles = {portrait, landscape};
export default styles;
