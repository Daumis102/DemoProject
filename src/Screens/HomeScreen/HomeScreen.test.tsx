import {fireEvent, render, screen} from '../../test-utils';
import {HomeScreen} from './HomeScreen';

describe('HomeScreen', () => {
  beforeEach(() => {
    render(<HomeScreen />);
  });
  it('should clear the input field after pressing add', async () => {
    const {inputField} = fillNameInputAndPress('foo');
    expect(inputField.props.value).toBe('');
  });
  it('should add a new items with a text entered in the input field after pressing add', async () => {
    fillNameInputAndPress('foo');
    fillNameInputAndPress('bar');

    expect(screen.getByText('foo')).toBeTruthy();
    expect(screen.getByText('bar')).toBeTruthy();
  });

  it('should match a snapshot after adding a new item to the list', () => {
    fillNameInputAndPress('foo');
    expect(screen.toJSON()).toMatchSnapshot();
  });

  function fillNameInputAndPress(text: string) {
    const inputField = screen.getByPlaceholderText('Enter name');
    fireEvent.changeText(inputField, text);

    const addButton = screen.getByText('Add');
    fireEvent.press(addButton);

    return {inputField, addButton};
  }
});
