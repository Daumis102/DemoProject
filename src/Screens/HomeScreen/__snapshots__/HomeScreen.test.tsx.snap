// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen should match a snapshot after adding a new item to the list 1`] = `
<View
  style={
    {
      "borderWidth": 1,
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      {
        "flexBasis": 150,
      }
    }
  >
    <Text
      style={
        {
          "flex": 1,
          "textAlign": "center",
          "textAlignVertical": "center",
        }
      }
    >
      Hello, enter names:
    </Text>
    <TextInput
      onChangeText={[Function]}
      placeholder="Enter name"
      style={
        {
          "borderRadius": 3,
          "borderWidth": 2,
          "flexBasis": 30,
          "margin": 5,
          "padding": 5,
        }
      }
      value=""
    />
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Add
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView
    style={
      {
        "borderWidth": 1,
      }
    }
  >
    <View>
      <View>
        <Text>
          foo
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
